### PR TITLE
integrate traderjoe v2.2

### PR DESCRIPTION
**Pull request by sunspirit9999**

### CODE REVIEW

File: pkg/source/liquiditybookv20/pool_tracker.go [hunk 1]
```diff
@@ -234,6 +234,15 @@ func (d *PoolTracker) querySubgraph(ctx context.Context, p entity.Pool) (*queryS
...
 		}
+
+		// if no bin returned, stop
+		if resp.Pair == nil || len(resp.Pair.Bins) == 0 {
+			logger.WithFields(logger.Fields{
+				"poolAddress": p.Address,
+			}).Info("no bin returned")
+			break
+		}
+
```
Comment (lehainam-dev) [inline pkg/source/liquiditybookv20/pool_tracker.go:245]: Can you explain more about this change?

Comment (sunspirit9999) [reply pkg/source/liquiditybookv20/pool_tracker.go:245]: we should check resp.Pair != nil before use its properties, btw we can avoid to be panic (it really happened when i tested on my local)

Comment (lehainam-dev) [reply pkg/source/liquiditybookv20/pool_tracker.go:245]: Oh I see. Yeah I agree.

Comment (sunspirit9999) [reply pkg/source/liquiditybookv20/pool_tracker.go:245]: maybe due to the subgraph is still syncing

**Review by lehainam-dev — APPROVED** [review]

**Pull request merged**